### ISSUE TITLE
[no-floating-promises] Rule proposal: Add ignore for typescript types or specific methods

### ISSUE THREAD

**Issue by SimonSimCity** [typescript-eslint/typescript-eslint#4791]
### Description

Some of the functions I have in my code do return a promise, where I'm fine having them "floating around". They usually are functions whose promise already is handled in a different way, and I don't want to add `.catch(() => {})` to all of them.

One example I have at hand is the package `@vue/apollo-composable` which lets me run a query to a GraphQL server. This function returns properties (result, error, loading) which are objects which will get the value as they arrive (vue calls it reactive). The function `refetch()` now returns a promise which contains the result when it arrives, but I don't need it because the reactive properties will reflect the values.

My idea was now if it would be possible to ignore those calls but not all promises ... Maybe by type (`Promise<ApolloQueryResult<TResult>>`) or maybe by the name of the function returning the promise.

As you see, I myself am not yet that sure about how the extra feature to the rule `no-floating-promises` could be implemented, but am rather open to discuss it to find a good option. If you see it not being feasible, it would be nice to know a bit more about the implementation here, to wrap my head around what would be possible instead.

### Fail

```ts
new Promise(() => { });
```

### Pass

```ts
import * as VueApolloComposable from "@vue/apollo-composable";
import gql from "graphql-tag";

const query = VueApolloComposable.useQuery<Record<string, string>, Record<string, string>>(
  gql`
    query Test() {
  test() {
    result
  }
}`,
  {},
  {}
);

// Returns a promise which is already resolved by the query.result-property
query.refetch();
```

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/4722

If you're using async/await you can just await the promise and that's it.

Otherwise use the `void` operator to signal you're intentionally floating the promise.
